### ISSUE TITLE
Change to layout not reflected on production site

### ISSUE THREAD

**Issue by mydoghasworms** [sinatra/sinatra#1311]
I have a website running Sinatra and Puma with Nginx proxying traffic via an upstream to the Sinatra app.

Now I have made a change in views/layout.erb. Because the Sinatra app is running in production mode, I have restarted both Puma and Nginx, but when refreshing the browser, I cannot see the change in the layout.

The settings are just standard; I have not made any out-of-the ordinary settings, especially not relating to caching. I have also looked to see if there are any directories containing caches under /var for Nginx, but there are none.

Because the page is rendered by Sinatra, however, I can't see how it can be Nginx or Puma; otherwise my other dynamic pages, rendered by Sinatra, would also suffer the same problem surely?

**Comment by rennex** [comment]
Restarting Puma reloads the Sinatra app, so the change should have taken effect. Usually Nginx doesn't even need to be restarted when the upstream app is restarted.

I would suggest the following steps for debugging:

- Sometimes people make edits with faulty logic, so they don't cause a change in the output even when the code was successfully reloaded. One way to check for sure is to add an html comment near the top (outside of any "if" statement or such), and check the page source to see if the changes are coming through.

- Check the page with another browser or `curl` to eliminate any browser cache situation.

- If you have multiple servers or other copies of the app sources, make sure that you're editing and restarting the one that your browser is pointed at!

- If nothing else works, look for things you CAN change successfully. If you edit a template or the app's Ruby code, do those changes work? Can you add or delete a file? Etc.

**Comment by mydoghasworms** [comment]
@rennex I had exhausted all those avenues.

However, I went and checked now and I noticed that, even after restarting puma, the process associated with its socket was always the same. So I killed that process, restarted puma, and now the updates are reflecting.

If it happens again, I will see if I can take the issue up with the Puma team. Maybe there is something in their documentation that I missed. Thanks for your help!
